core/domain/rights_manager.py

```diff
@@ -625,18 +625,15 @@ def _change_activity_status(
 
     if new_status != ACTIVITY_STATUS_PRIVATE:
         activity_rights.viewer_ids = []
+        if activity_rights.first_published_msec is None:
```

Comment:
Oh, wow, good catch. Looks like we might have been computing this value incorrectly for a while...